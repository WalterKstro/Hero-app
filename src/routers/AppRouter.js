import { useContext } from "react";
import {BrowserRouter as Router,Switch} from "react-router-dom";
import AuthContext from "../auth/AuthContext";
import LoginPage from "../componentes/login/LoginPage";
import AuthenticatedRouters from "./AuthenticatedRouters";
import PrivateRouter from "./PrivateRouter";
import PublicRouter from "./PublicRouter";

const AppRouter = () => {
    const {user} = useContext(AuthContext)
    
    return (
        <Router>
            <header>
                <Switch>
                    <PublicRouter 
                        exact 
                        path="/login" 
                        component={LoginPage}
                        isAuthenticated={user.logged}
                    />

                    <PrivateRouter 
                        path="/" 
                        component={AuthenticatedRouters}
                        isAuthenticated={user.logged}
                    />
                </Switch>
            </header>
        </Router>
    )
}

export default AppRouter
