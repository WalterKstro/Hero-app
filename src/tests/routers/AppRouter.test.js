import { mount } from "enzyme"
import AuthContext from "../../auth/AuthContext"
import AppRouter from "../../routers/AppRouter"

describe('Testing the <AppRouter/>', () => {

    test("should show the page login, if it's authenticated", () => {
        const user = {logged: false}
        const action = jest.fn()

        const rendered = mount(
            <AuthContext.Provider value={{user,action}}>
                <AppRouter/>
            </AuthContext.Provider>
        )

        expect(rendered.find('input').length).toBe(2)
        expect(rendered).toMatchSnapshot()
    })


    test("should show the page home if it's authenticated", () => {
        const user = {
            logged: true,
            email: 'Walter',
            pass: '123'
        }
        const action = jest.fn()

        const rendered = mount(
            <AuthContext.Provider value={{user,action}}>
                <AppRouter/>
            </AuthContext.Provider>
        )
        
        expect(rendered.find('h1').text()).toBe('Frontend Developer')
        expect(rendered).toMatchSnapshot()
    })
    
    
})
