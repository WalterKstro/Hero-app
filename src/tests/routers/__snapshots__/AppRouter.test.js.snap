// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <AppRouter/> should show the page home if it's authenticated 1`] = `
<header>
  Array [
    <nav
      className="sticky-top navbar navbar-expand-lg navbar-dark bg-dark py-0"
    >
      <div
        className="container-fluid"
      >
        <a
          aria-current="page"
          className="navbar-brand active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <img
            alt="Brand"
            src="/assets/Brand.svg"
          />
        </a>
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-bs-target="#navbarNav"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            className="navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/dc"
                onClick={[Function]}
              >
                DC Universe
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/marvel"
                onClick={[Function]}
              >
                Marvel Universe
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/search"
                onClick={[Function]}
              >
                Search
              </a>
            </li>
          </ul>
        </div>
        <div
          className="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            className="navbar-nav ms-auto"
          >
            <li
              className="nav-item"
            >
              <button
                className="nav-link text-white fw-bold btn"
              >
                Walter
              </button>
            </li>
            <li
              className="nav-item"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
              >
                Sign Out
              </button>
            </li>
          </ul>
        </div>
      </div>
    </nav>,
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="home col d-flex flex-md-row flex-sm-column align-items-center justify-content-between"
        >
          <main>
            <h1>
              Frontend Developer
            </h1>
            <address>
              <strong>
                Walter Castro
              </strong>
              <strong
                className="d-block"
              >
                <a
                  href="mailto:walterkstro@gmail.com"
                >
                  walterkstro@gmail.com
                </a>
              </strong>
              <p>
                City Guatemala
              </p>
            </address>
          </main>
          <figure>
            <img
              alt="Code QR LinkedIn"
              src="/assets/linkedin.jpeg"
            />
          </figure>
        </div>
      </div>
    </div>,
  ]
</header>
`;

exports[`Testing the <AppRouter/> should show the page login, if it's authenticated 1`] = `
<header>
  <div
    className="container"
  >
    <div
      className="row my-5 justify-content-center"
    >
      <div
        className="col-4 col-md-2 col-md-offset-5"
      >
        <img
          alt="Brand"
          src="/assets/Brand.svg"
        />
      </div>
    </div>
    <div
      className="row justify-content-center"
    >
      <div
        className="col-12 col-md-6"
      >
        <form
          onSubmit={[Function]}
        >
          <div>
            <label
              className="col-sm-2 col-form-label"
              htmlFor="staticUsername"
            >
              Username
            </label>
            <div
              className="col-sm-10"
            >
              <input
                className="form-control"
                id="staticEmail"
                name="email"
                onChange={[Function]}
                placeholder="your username"
                type="text"
              />
            </div>
            <div
              className="form-text"
            >
              Any username
            </div>
          </div>
          <div
            className="mb-3"
          >
            <label
              className="col-sm-2 col-form-label"
              htmlFor="inputPassword"
            >
              Password
            </label>
            <div
              className="col-sm-10"
            >
              <input
                className="form-control"
                id="inputPassword"
                name="pass"
                onChange={[Function]}
                placeholder="********"
                type="password"
              />
              <div
                className="form-text"
              >
                Any password
              </div>
            </div>
          </div>
          <div>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Log In
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</header>
`;
